### ISSUE TITLE
Protocol woes

### ISSUE THREAD

**Issue by algernon** [keyboardio/KeyboardioHID#38]
We have a problem.

With the recent change to default `BootKeyboard` to boot protocol, Linux never sets it to report protocol, thus, the keyboard will not be NKRO, only 6KRO.

If we default to report protocol, then wakeup won't work, and certain BIOSes may end up not liking it, either.

We need to spend a bit more time on this, to figure out if there's anything at all we can do, to support BIOSes, NKRO and wakeup automatically, reliably, across all major operating systems. There are a couple of tricks we could try, so not everything is lost yet.

**Comment by algernon** [comment]
See [here](https://deskthority.net/wiki/USB#Workarounds) for a possible workaround.

The current plan is to try the workaround **and** provide a way to force boot protocol during keyboard boot: by pressing `Left Fn + Cmd + Esc` at boot, the keyboard would default to boot protocol. For this to work, we need to move the `BootKeyboard` plugging from the constructor to somewhere else, where we can do a scan first, and init BootKeyboard based on the results.

**Comment by algernon** [comment]
Tried the workaround, and it works, for some values of working. Linux correctly interprets the case, there are no double keystrokes or anything like that. However, input becomes unusably slow for some reason.

Thus, it appears that the workaround is not suitable for our purposes. Will give myself another hour or so to try and figure out why Linux gets so slow, but it's not worth spending more time on it than that.

**Comment by algernon** [comment]
It turns out that if I set all reported parts to `D_CONSTANT`, sending the report will become incredibly slow - possibly because the OS doesn't care much, and doesn't respond fast enough. However, if I keep at least one part of it `D_DATA`, the latency is as usual, and the keyboard works fine under Linux.

Happy that this might work, I tested under OSX: FileVault worked wonderfully, but nothing after: the `BootKeyboard` with constants is not ignored by OSX, and we get duplicate keys.
